plenum/server/replica.py

```diff
@@ -774,9 +776,19 @@ def processReqDuringBatch(
             self.logger.warning('{} encountered exception {} while processing {}, '
                                 'will reject'.format(self, ex, req))
             rejects.append((req.key, Reject(req.identifier, req.reqId, ex)))
-            inValidReqs.append(req)
+            discarded_mask.append(False)
         else:
-            validReqs.append(req)
+            discarded_mask.append(True)
+        finally:
+            reqs.append(req)
+
+    def _pack_discarded_mask(self, value):
```

Comment:
Please add unit tests for this 2 methods. Should we make the methods static, and maybe even move to some utils?